clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/cluster/AzureClusterProvider.groovy

```diff
@@ -0,0 +1,233 @@
+      serverGroups = translateServerGroups(allServerGroups)
+    }
+
+    Collection<AzureCluster> clusters = clusterData.collect { CacheData clusterDataEntry ->
+      Map<String, String> clusterKey = Keys.parse(azureCloudProvider, clusterDataEntry.id)
+
+      def cluster = new AzureCluster()
+      cluster.accountName = clusterKey.account
+      cluster.name = clusterKey.name
+      cluster.type = azureCloudProvider.id
+      if (includeDetails) {
+        cluster.loadBalancers = clusterDataEntry.relationships[AZURE_LOAD_BALANCERS.ns]?.findResults {
+          loadBalancers.get(it)
+        }
+        cluster.serverGroups = clusterDataEntry.relationships[AZURE_SERVER_GROUPS.ns]?.findResults {
+          serverGroups.get(it)
+        }
+      } else {
+        cluster.loadBalancers = clusterDataEntry.relationships[AZURE_LOAD_BALANCERS.ns]?.collect { loadBalancerKey ->
+          Map parts = Keys.parse(azureCloudProvider, loadBalancerKey)
+          new AzureLoadBalancer(name: parts.name, region: parts.region, account: parts.account)
+        }
+
+        cluster.serverGroups = clusterDataEntry.relationships[AZURE_SERVER_GROUPS.ns]?.collect { serverGroupKey ->
+          Map parts = Keys.parse(azureCloudProvider, serverGroupKey)
+          new AzureServerGroupDescription(name: parts.serverGroup, region: parts.region,
+            application: parts.application, appName: parts.application)
+        }
+      }
+      cluster
+    }
+
+    clusters
+  }
+
+  private Map<String, AzureLoadBalancer> translateLoadBalancers(Collection<CacheData> loadBalancerData) {
+    loadBalancerData.collectEntries { loadBalancerEntry ->
+      Map<String, String> lbKey = Keys.parse(azureCloudProvider, loadBalancerEntry.id)
+      [(loadBalancerEntry.id) : new AzureLoadBalancer(name: lbKey.name, account: lbKey.account, region: lbKey.region)]
+    }
+  }
+
+  private Map<String, AzureServerGroupDescription> translateServerGroups(Collection<CacheData> serverGroupData) {
+
+    Map<String, AzureServerGroupDescription> serverGroups = serverGroupData.collectEntries { serverGroupEntry ->
+
+      def serverGroup = objectMapper.convertValue(serverGroupEntry.attributes.serverGroup, AzureServerGroupDescription)
+      def instances = resolveRelationshipData(serverGroupEntry, AZURE_INSTANCES.ns)
+      serverGroup.instances = translateInstances(instances)
+      [(serverGroupEntry.id): serverGroup]
+    }
+
+    serverGroups
+  }
+
+  private static Set<AzureInstance> translateInstances(Collection<CacheData> instanceData) {
+    def instances = instanceData?.collect { instanceEntry ->
+      AzureInstance instance = null
+      try {
+        instance = new AzureInstance(instanceEntry.attributes.instance)
+        //instance.name = instanceEntry.attributes.name.toString()
+      } catch (Exception e) {
+        def exception = e
+      }
+      instance
+    } ?: []
+    //addHealthToInstances(instanceData, instances)
```

Comment:
If the commented out code is stuff that will be added a little later please decorate it with a TODO as well as an indicator, otherwise delete.